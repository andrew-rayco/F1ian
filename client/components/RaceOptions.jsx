import React from 'react'
import moment from 'moment'

const RaceOptions = ({quali, visibility}) => {

  let hidden
  (visibility) ? hidden = 'hidden' : hidden = ''

  function toggleRound(e, raceId) {
    e.preventDefault()
    let raceDetails = document.getElementById(raceId)
    raceDetails.classList.toggle('hidden')
  }

  return (
    <div key={quali.raceId} className="row single-round">
      <div className="twelve columns round">
        <h4 onClick={(e) => toggleRound(e, quali.raceId)}><a href="#">Round {quali.round} - {quali.raceName}</a></h4>
        <div id={quali.raceId} className={`toggle ${hidden}`}>
          <p>{moment(quali.date).format('MMMM Do YYYY')}</p>
          <p><a href={`/#/season/${quali.year}/${quali.raceId}/qualifying`}>Qualifying results</a></p>
          <p><a href={`/#/season/${quali.year}/${quali.raceId}/grid`}>Starting grid</a></p>
          <p><a href={`/#/season/${quali.year}/${quali.raceId}/visualise`}>Visualise</a></p>
          <p><a href={`/#/season/${quali.year}/${quali.raceId}/results`}>Results</a></p>
          <p><a href={quali.raceUrl}>{quali.year} {quali.raceName} on Wikipedia</a></p>
          <div className="separator"></div>
        </div>
      </div>
    </div>
  )
}

module.exports = RaceOptions
