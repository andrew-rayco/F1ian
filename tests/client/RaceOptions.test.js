import React from 'react'
import { shallow } from 'enzyme'
import RaceOptions from '../../client/components/RaceOptions'

describe('RaceOptions component', () => {
  const props = {
    race: {
      raceId: 999
    }
  }
  const app = shallow(<RaceOptions race={props} />)

  it('renders correctly', () => {
    expect(app).toMatchSnapshot()
  })
})
