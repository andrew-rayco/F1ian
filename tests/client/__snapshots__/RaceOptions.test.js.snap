// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RaceOptions component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RaceOptions
    props={
      Object {
        "race": Object {
          "raceId": 999,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="twelve columns round undefined"
      >
        <h4
          onClick={[Function]}
        >
          <a
            href="#"
          >
            undefined - undefined
          </a>
        </h4>
        <div
          className="toggle hidden"
          id={undefined}
        >
          <p>
            December 8th 2017
          </p>
          <p>
            <a
              className="togglable"
              href="#"
              onClick={[Function]}
            >
              Qualifying results 
              <img
                alt="read more icon"
                className="toggle-icon"
                src="../images/down-arrow.svg"
              />
            </a>
          </p>
          <p>
            <a
              className="togglable"
              href="#"
              onClick={[Function]}
            >
              Starting grid 
              <img
                alt="read more icon"
                className="toggle-icon"
                src="../images/down-arrow.svg"
              />
            </a>
          </p>
          <p>
            <a
              href="/#/season/undefined/undefined/visualise"
            >
              Visualise
            </a>
          </p>
          <p>
            <a
              className="togglable"
              href="#"
              onClick={[Function]}
            >
              Results 
              <img
                alt="read more icon"
                className="toggle-icon"
                src="../images/down-arrow.svg"
              />
            </a>
          </p>
          <p>
            <a
              href={undefined}
            >
               
               on Wikipedia
            </a>
          </p>
          <div
            className="separator"
          />
        </div>
      </div>,
      "className": "row single-round",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4
            onClick={[Function]}
          >
            <a
              href="#"
            >
              undefined - undefined
            </a>
          </h4>,
          <div
            className="toggle hidden"
            id={undefined}
          >
            <p>
              December 8th 2017
            </p>
            <p>
              <a
                className="togglable"
                href="#"
                onClick={[Function]}
              >
                Qualifying results 
                <img
                  alt="read more icon"
                  className="toggle-icon"
                  src="../images/down-arrow.svg"
                />
              </a>
            </p>
            <p>
              <a
                className="togglable"
                href="#"
                onClick={[Function]}
              >
                Starting grid 
                <img
                  alt="read more icon"
                  className="toggle-icon"
                  src="../images/down-arrow.svg"
                />
              </a>
            </p>
            <p>
              <a
                href="/#/season/undefined/undefined/visualise"
              >
                Visualise
              </a>
            </p>
            <p>
              <a
                className="togglable"
                href="#"
                onClick={[Function]}
              >
                Results 
                <img
                  alt="read more icon"
                  className="toggle-icon"
                  src="../images/down-arrow.svg"
                />
              </a>
            </p>
            <p>
              <a
                href={undefined}
              >
                 
                 on Wikipedia
              </a>
            </p>
            <div
              className="separator"
            />
          </div>,
        ],
        "className": "twelve columns round undefined",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#"
            >
              undefined - undefined
            </a>,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "undefined - undefined",
              "href": "#",
            },
            "ref": null,
            "rendered": "undefined - undefined",
            "type": "a",
          },
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                December 8th 2017
              </p>,
              <p>
                <a
                  className="togglable"
                  href="#"
                  onClick={[Function]}
                >
                  Qualifying results 
                  <img
                    alt="read more icon"
                    className="toggle-icon"
                    src="../images/down-arrow.svg"
                  />
                </a>
              </p>,
              null,
              <p>
                <a
                  className="togglable"
                  href="#"
                  onClick={[Function]}
                >
                  Starting grid 
                  <img
                    alt="read more icon"
                    className="toggle-icon"
                    src="../images/down-arrow.svg"
                  />
                </a>
              </p>,
              null,
              <p>
                <a
                  href="/#/season/undefined/undefined/visualise"
                >
                  Visualise
                </a>
              </p>,
              <p>
                <a
                  className="togglable"
                  href="#"
                  onClick={[Function]}
                >
                  Results 
                  <img
                    alt="read more icon"
                    className="toggle-icon"
                    src="../images/down-arrow.svg"
                  />
                </a>
              </p>,
              null,
              <p>
                <a
                  href={undefined}
                >
                   
                   on Wikipedia
                </a>
              </p>,
              <div
                className="separator"
              />,
            ],
            "className": "toggle hidden",
            "id": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "December 8th 2017",
              },
              "ref": null,
              "rendered": "December 8th 2017",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="togglable"
                  href="#"
                  onClick={[Function]}
                >
                  Qualifying results 
                  <img
                    alt="read more icon"
                    className="toggle-icon"
                    src="../images/down-arrow.svg"
                  />
                </a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Qualifying results ",
                    <img
                      alt="read more icon"
                      className="toggle-icon"
                      src="../images/down-arrow.svg"
                    />,
                  ],
                  "className": "togglable",
                  "href": "#",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  "Qualifying results ",
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "read more icon",
                      "className": "toggle-icon",
                      "src": "../images/down-arrow.svg",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                ],
                "type": "a",
              },
              "type": "p",
            },
            null,
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="togglable"
                  href="#"
                  onClick={[Function]}
                >
                  Starting grid 
                  <img
                    alt="read more icon"
                    className="toggle-icon"
                    src="../images/down-arrow.svg"
                  />
                </a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Starting grid ",
                    <img
                      alt="read more icon"
                      className="toggle-icon"
                      src="../images/down-arrow.svg"
                    />,
                  ],
                  "className": "togglable",
                  "href": "#",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  "Starting grid ",
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "read more icon",
                      "className": "toggle-icon",
                      "src": "../images/down-arrow.svg",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                ],
                "type": "a",
              },
              "type": "p",
            },
            null,
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="/#/season/undefined/undefined/visualise"
                >
                  Visualise
                </a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Visualise",
                  "href": "/#/season/undefined/undefined/visualise",
                },
                "ref": null,
                "rendered": "Visualise",
                "type": "a",
              },
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="togglable"
                  href="#"
                  onClick={[Function]}
                >
                  Results 
                  <img
                    alt="read more icon"
                    className="toggle-icon"
                    src="../images/down-arrow.svg"
                  />
                </a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Results ",
                    <img
                      alt="read more icon"
                      className="toggle-icon"
                      src="../images/down-arrow.svg"
                    />,
                  ],
                  "className": "togglable",
                  "href": "#",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  "Results ",
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "read more icon",
                      "className": "toggle-icon",
                      "src": "../images/down-arrow.svg",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                ],
                "type": "a",
              },
              "type": "p",
            },
            null,
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href={undefined}
                >
                   
                   on Wikipedia
                </a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    undefined,
                    " ",
                    undefined,
                    " on Wikipedia",
                  ],
                  "href": undefined,
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                  " ",
                  undefined,
                  " on Wikipedia",
                ],
                "type": "a",
              },
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "className": "separator",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="twelve columns round undefined"
        >
          <h4
            onClick={[Function]}
          >
            <a
              href="#"
            >
              undefined - undefined
            </a>
          </h4>
          <div
            className="toggle hidden"
            id={undefined}
          >
            <p>
              December 8th 2017
            </p>
            <p>
              <a
                className="togglable"
                href="#"
                onClick={[Function]}
              >
                Qualifying results 
                <img
                  alt="read more icon"
                  className="toggle-icon"
                  src="../images/down-arrow.svg"
                />
              </a>
            </p>
            <p>
              <a
                className="togglable"
                href="#"
                onClick={[Function]}
              >
                Starting grid 
                <img
                  alt="read more icon"
                  className="toggle-icon"
                  src="../images/down-arrow.svg"
                />
              </a>
            </p>
            <p>
              <a
                href="/#/season/undefined/undefined/visualise"
              >
                Visualise
              </a>
            </p>
            <p>
              <a
                className="togglable"
                href="#"
                onClick={[Function]}
              >
                Results 
                <img
                  alt="read more icon"
                  className="toggle-icon"
                  src="../images/down-arrow.svg"
                />
              </a>
            </p>
            <p>
              <a
                href={undefined}
              >
                 
                 on Wikipedia
              </a>
            </p>
            <div
              className="separator"
            />
          </div>
        </div>,
        "className": "row single-round",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              onClick={[Function]}
            >
              <a
                href="#"
              >
                undefined - undefined
              </a>
            </h4>,
            <div
              className="toggle hidden"
              id={undefined}
            >
              <p>
                December 8th 2017
              </p>
              <p>
                <a
                  className="togglable"
                  href="#"
                  onClick={[Function]}
                >
                  Qualifying results 
                  <img
                    alt="read more icon"
                    className="toggle-icon"
                    src="../images/down-arrow.svg"
                  />
                </a>
              </p>
              <p>
                <a
                  className="togglable"
                  href="#"
                  onClick={[Function]}
                >
                  Starting grid 
                  <img
                    alt="read more icon"
                    className="toggle-icon"
                    src="../images/down-arrow.svg"
                  />
                </a>
              </p>
              <p>
                <a
                  href="/#/season/undefined/undefined/visualise"
                >
                  Visualise
                </a>
              </p>
              <p>
                <a
                  className="togglable"
                  href="#"
                  onClick={[Function]}
                >
                  Results 
                  <img
                    alt="read more icon"
                    className="toggle-icon"
                    src="../images/down-arrow.svg"
                  />
                </a>
              </p>
              <p>
                <a
                  href={undefined}
                >
                   
                   on Wikipedia
                </a>
              </p>
              <div
                className="separator"
              />
            </div>,
          ],
          "className": "twelve columns round undefined",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="#"
              >
                undefined - undefined
              </a>,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "undefined - undefined",
                "href": "#",
              },
              "ref": null,
              "rendered": "undefined - undefined",
              "type": "a",
            },
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  December 8th 2017
                </p>,
                <p>
                  <a
                    className="togglable"
                    href="#"
                    onClick={[Function]}
                  >
                    Qualifying results 
                    <img
                      alt="read more icon"
                      className="toggle-icon"
                      src="../images/down-arrow.svg"
                    />
                  </a>
                </p>,
                null,
                <p>
                  <a
                    className="togglable"
                    href="#"
                    onClick={[Function]}
                  >
                    Starting grid 
                    <img
                      alt="read more icon"
                      className="toggle-icon"
                      src="../images/down-arrow.svg"
                    />
                  </a>
                </p>,
                null,
                <p>
                  <a
                    href="/#/season/undefined/undefined/visualise"
                  >
                    Visualise
                  </a>
                </p>,
                <p>
                  <a
                    className="togglable"
                    href="#"
                    onClick={[Function]}
                  >
                    Results 
                    <img
                      alt="read more icon"
                      className="toggle-icon"
                      src="../images/down-arrow.svg"
                    />
                  </a>
                </p>,
                null,
                <p>
                  <a
                    href={undefined}
                  >
                     
                     on Wikipedia
                  </a>
                </p>,
                <div
                  className="separator"
                />,
              ],
              "className": "toggle hidden",
              "id": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "December 8th 2017",
                },
                "ref": null,
                "rendered": "December 8th 2017",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    className="togglable"
                    href="#"
                    onClick={[Function]}
                  >
                    Qualifying results 
                    <img
                      alt="read more icon"
                      className="toggle-icon"
                      src="../images/down-arrow.svg"
                    />
                  </a>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Qualifying results ",
                      <img
                        alt="read more icon"
                        className="toggle-icon"
                        src="../images/down-arrow.svg"
                      />,
                    ],
                    "className": "togglable",
                    "href": "#",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Qualifying results ",
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "alt": "read more icon",
                        "className": "toggle-icon",
                        "src": "../images/down-arrow.svg",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                  ],
                  "type": "a",
                },
                "type": "p",
              },
              null,
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    className="togglable"
                    href="#"
                    onClick={[Function]}
                  >
                    Starting grid 
                    <img
                      alt="read more icon"
                      className="toggle-icon"
                      src="../images/down-arrow.svg"
                    />
                  </a>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Starting grid ",
                      <img
                        alt="read more icon"
                        className="toggle-icon"
                        src="../images/down-arrow.svg"
                      />,
                    ],
                    "className": "togglable",
                    "href": "#",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Starting grid ",
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "alt": "read more icon",
                        "className": "toggle-icon",
                        "src": "../images/down-arrow.svg",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                  ],
                  "type": "a",
                },
                "type": "p",
              },
              null,
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    href="/#/season/undefined/undefined/visualise"
                  >
                    Visualise
                  </a>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Visualise",
                    "href": "/#/season/undefined/undefined/visualise",
                  },
                  "ref": null,
                  "rendered": "Visualise",
                  "type": "a",
                },
                "type": "p",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    className="togglable"
                    href="#"
                    onClick={[Function]}
                  >
                    Results 
                    <img
                      alt="read more icon"
                      className="toggle-icon"
                      src="../images/down-arrow.svg"
                    />
                  </a>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Results ",
                      <img
                        alt="read more icon"
                        className="toggle-icon"
                        src="../images/down-arrow.svg"
                      />,
                    ],
                    "className": "togglable",
                    "href": "#",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Results ",
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "alt": "read more icon",
                        "className": "toggle-icon",
                        "src": "../images/down-arrow.svg",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                  ],
                  "type": "a",
                },
                "type": "p",
              },
              null,
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    href={undefined}
                  >
                     
                     on Wikipedia
                  </a>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      undefined,
                      " ",
                      undefined,
                      " on Wikipedia",
                    ],
                    "href": undefined,
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                    " ",
                    undefined,
                    " on Wikipedia",
                  ],
                  "type": "a",
                },
                "type": "p",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "className": "separator",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
